mdoc/src/main/scala-3/mdoc/internal/cli/Settings.scala

```diff
@@ -0,0 +1,121 @@
+package mdoc.internal.cli
+
+import java.io.InputStream
+import java.nio.charset.Charset
+import java.nio.charset.IllegalCharsetNameException
+import java.nio.charset.StandardCharsets
+import java.nio.charset.UnsupportedCharsetException
+import java.nio.file.FileSystems
+import java.nio.file.Files
+import java.nio.file.InvalidPathException
+import java.nio.file.PathMatcher
+import mdoc.OnLoadContext
+import metaconfig.Conf
+import metaconfig.ConfDecoder
+import metaconfig.ConfEncoder
+import metaconfig.ConfError
+import metaconfig.Configured
+import metaconfig.annotation._
+import metaconfig.generic
+import metaconfig.generic.Surface
+import org.typelevel.paiges.Doc
+import scala.annotation.StaticAnnotation
+import scala.meta.internal.io.PathIO
+import scala.meta.io.AbsolutePath
+import scala.meta.io.RelativePath
+import mdoc.Reporter
+import mdoc.internal.markdown.MarkdownCompiler
+
+class Section(val name: String) extends StaticAnnotation
+
+case class Settings(
```

Comment:
This is also greatly stripped down version, msotly due to the fact that most settings were used for CLI, which is not yet needed for Scala 3